Add docs for fleetctl get mdm-commands (#12006)
fleetdm/fleet: docs/Using-Fleet/MDM-commands.md

1. Run `fleetctl get mdm-commands`
2. View the list of recently executed commands, most recent first, along with the timestamp, targeted hostname, command type, execution status and command ID.

Example output:

```
$ fleetctl get mdm-commands
+--------------------------------------+----------------------+--------------------------+--------------+------------------------+
|                  ID                  |         TIME         |           TYPE           |    STATUS    |        HOSTNAME        |
+--------------------------------------+----------------------+--------------------------+--------------+------------------------+
| 024fb3b9-cd8a-40a6-8dd7-6c155f488fd1 | 2023-04-12T18:19:10Z | InstalledApplicationList | Acknowledged | iMac-Pro.local         |
+--------------------------------------+----------------------+--------------------------+--------------+------------------------+
| 87dc6325-8bc0-4fc8-9a2f-3901c535456e | 2023-04-12T18:15:01Z | InstallProfile           | Acknowledged | iMac-Pro.local         |
+--------------------------------------+----------------------+--------------------------+--------------+------------------------+
```

> The command ID can be used to view command results as documented in [Step 4 of the previous section](#step-4-view-the-commands-results). The possible status values are:

Reviewer: ```suggestion
The command ID can be used to view command results as documented in [step 4 of the previous section](#step-4-view-the-commands-results). The possible status values are:
```

Small nit

2. Look at the on-screen information.

> ## List of recent commands

Reviewer: ```suggestion
## List recent commands
```

Minor nit. I think we can make "list" a verb here. This communicates to the user that they should follow this guide if they want to list recent commands.

You can view the list of recently executed commands using "fleetctl":

1. Run `fleetctl get mdm-commands`
2. View the list of recently executed commands, most recent first, along with the timestamp, targeted hostname, command type, execution status and command ID.

> 

Reviewer: @mna thanks! Do you know what the code blocks for the example will look like when they get deployed to the website?

I want to make sure that the tables don't look wonky on fleetdm.com/docs.

There's some way to test doc changes on the website locally but I can't remember the sails commands to do this? @eashaw can you please remind us? Thanks

> 

Author: I ran the website locally , they look ok to me, only thing is the end of the command results table, but that's how the output from fleetctl is, not an issue with the rendering of the docs (I think it's the 3rd-party library that we use to format tables that causes this):

---
Start of get mdm-command-results:

![Screenshot from 2023-05-31 10-07-46](https://github.com/fleetdm/fleet/assets/692384/c1d52ce6-d923-4538-8be5-c15ba1f2cef7)

---

End of get mdm-command-results:

![Screenshot from 2023-05-31 10-07-55](https://github.com/fleetdm/fleet/assets/692384/52d8fe20-ad54-4774-ab60-5bb84e477927)

---

Start of get mdm-commands:

![Screenshot from 2023-05-31 10-08-04](https://github.com/fleetdm/fleet/assets/692384/d162946b-6275-4fa9-9de6-9d70cae98cb5)

---

End of get mdm-commands:

![Screenshot from 2023-05-31 10-08-12](https://github.com/fleetdm/fleet/assets/692384/6a918d51-d371-49e8-8394-0e8d6d156dd5)

> 

Reviewer: Yeah...the end of `get mdm-command-results` could be improved. Fleet looks broken. @mna when you get the chance, can you please file a bug report issue for this?

The command ID can be used to view command results as documented in [Step 4 of the previous section](#step-4-view-the-commands-results). The possible status values are:

> * Pending: the command has yet to be executed on the host.

Reviewer: ```suggestion
* Pending: the command has yet to run on the host. The host will run the command the next time it comes online.
```

Minor nit. We're using "run" language elsewhere in the docs. I think we should be consistent.

Also I added a bit of context to let the user know when the command will run.

> * Acknowledged: the command was executed successfully.

Reviewer: ```suggestion
* Acknowledged: the host responded with "Acknowledged" status via the MDM protocol: the host processed the command successfully.
```

I think we should be more explicit here about how "Acknowledged" is a status that comes back from the MDM protocol. A lot of users express problems of getting Acknowledged back and not really knowing if the command ran or not.

I also borrowed Apple's [definition of Acknowledged](https://developer.apple.com/documentation/devicemanagement/installprofileresponse): 

![Screenshot 2023-05-30 at 4 21 39 PM](https://github.com/fleetdm/fleet/assets/47070608/2f75e4ca-78dc-46f7-93dc-753ca6157c2d)

> * Error: execution of the command failed.

Reviewer: ```suggestion
* Error: the host responded with "Error" status via the MDM protocol: an error occurred. Run the `fleetctl get mdm-command-results --id=<insert-command-id` to view the error.
```

I think we should be explicit here too. We're taking the status from the MDM protocol right?

Author: Yes, that's correct.
